File: hw/santorum/u3/hw29Singleton/JavaScript/controller/AppointmentController.js
import StorageManager from "../utils/StorageManager.js";
import EmailService from "../utils/EmailService.js";

export default class AppointmentController {
  async registerAppointment(appointment) {
    StorageManager.saveAppointment(appointment);

    await EmailService.sendEmail(
      appointment.patientEmail,
      "Your appointment has been registered successfully."
    );
  }
}

File: hw/santorum/u3/hw29Singleton/Python/controller/appointment_controller.py
from utils.file_manager import FileManager
from utils.email_service import EmailService

class AppointmentController:

    def register_appointment(self, appointment):
        FileManager.save_appointment(appointment)

        EmailService.send_email(
            appointment.patient_email,
            "Your appointment has been registered successfully."
        )
